connectors/sources/s3.py

```diff
@@ -60,17 +62,34 @@ def __init__(self, configuration):
             retries={"max_attempts": self.configuration["max_attempts"]},
         )
         self.enable_content_extraction = self.configuration["enable_content_extraction"]
+        self.s3_region_client = {}
+        self.s3_context_stacks = []
+
+    async def client(self, region=None):
+        """This method creates context manager and client session object for s3.
+        Args:
+            region (str): Name of bucket region. Defaults to None
```

Comment:
@seanstory In s3 it's mentioned [here](https://github.com/boto/boto3/blob/develop/boto3/session.py#L240) that a single client is associated with single region. Also I tried running S3 connector with single client but this will throw access denied error if bucket are in different regions.

Reply:
Looking at the examples, I see that it is (or was) possible to not pass any region inside of the client and still be able to access all buckets. Is it not the case any more?

Reply:
@artem-shelkovnikov For fetching list of all buckets we don't need to pass region but for accessing objects from those bucket we need to pass region..

Reply:
@akanshi-elastic I don't think that's right. I can see how the line you linked from their source code might imply that, but all the documentation I'm looking at shows that you don't need to specify a region: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-example-download-file.html

Reply:
As discussed with @artem-shelkovnikov we are not able to create a client without specifying the region with it. We've also raised an issue on GitHub for the same: https://github.com/aio-libs/aiobotocore/issues/1005.  So as of now If everything looks good we can merge this PR and we'll create an enhancement to separate client class.

Reply:
@akanshi-elastic can you follow up on https://github.com/aio-libs/aiobotocore/issues/1005#issuecomment-1512371598? Sounds like they are still not able to produce the issue you were having with the region, and this makes me hopeful that it's an issue with your local setup.

Reply:
we have taken a follow up and we will be scheduling a call with them.